test/e2e/far_e2e_test.go

```diff
@@ -0,0 +1,171 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/medik8s/fence-agents-remediation/api/v1alpha1"
+	farController "github.com/medik8s/fence-agents-remediation/controllers"
+	"github.com/medik8s/fence-agents-remediation/pkg/cli"
+	farUtils "github.com/medik8s/fence-agents-remediation/test/e2e/utils"
+)
+
+const (
+	testNamespace  = "fence-agents-remediation"
+	validNodeName  = "worker-0"
+	fenceAgentIPMI = "fence_ipmilan"
+	hostNameLabel  = "kubernetes.io/hostname"
+
+	// eventually parameters
+	timeout      = 2 * time.Minute
+	pollInterval = 10 * time.Second
+	offsetExpect = 1
+)
+
+var _ = Describe("FAR E2e", func() {
+	testShareParam := map[v1alpha1.ParameterName]string{
+		"--username": "admin",
+		"--password": "password",
+		"--action":   "reboot",
+		"--ip":       "192.168.111.1",
+		"--lanplus":  "",
+	}
+	testNodeParam := map[v1alpha1.ParameterName]map[v1alpha1.NodeName]string{
+		"--ipport": {
+			"master-0": "6230",
+			"master-1": "6231",
+			"master-2": "6232",
+			"worker-0": "6233",
+			"worker-1": "6234",
+			"worker-2": "6235",
+		},
+	} // get ports
+	Context("fence agent - fence_ipmilan", func() {
+		var testNode *corev1.Node
+		var far *v1alpha1.FenceAgentsRemediation
+		var testStart time.Time
+		oldNodes := &corev1.NodeList{}
+		req, _ := labels.NewRequirement(hostNameLabel, selection.Exists, []string{validNodeName})
+		selector := labels.NewSelector().Add(*req)
+		BeforeEach(func() {
+			Expect(k8sClient.List(context.Background(),
+				oldNodes, &client.ListOptions{LabelSelector: selector})).ToNot(HaveOccurred())
+			testNode = &oldNodes.Items[0]
+
+			// save boot time
+			testStart = time.Now()
+			far = createFAR(validNodeName, fenceAgentIPMI, testShareParam, testNodeParam)
```

Comment:
Nit: I think it could be a nice opportunity to use [DeferCleanup()](https://onsi.github.io/ginkgo/#cleaning-up-our-cleanup-code-defercleanup)

Reply:
Sounds like a good idea.
Since I am using an old Ginkgo version ([1.16.5](https://github.com/medik8s/fence-agents-remediation/blob/main/Makefile#L18)), then I can't use `DeferCleanup()`. I will give it a try on a follow up PR after this PR is merged and I will update all the of FAR tools (including Ginkgo).

Reply:
This PR takes longer time than expected, and Ginkgo v2 was already updated - #50

[DeferCleanup()](https://onsi.github.io/ginkgo/#cleaning-up-our-cleanup-code-defercleanup) was added to E2E test.